test/RaffleStore_test.js

```diff
@@ -93,18 +100,144 @@ describe("RaffleStore", function () {
     ).to.be.reverted
 
     await expect(
-      raffleStore.connect(rafflePlayer).enterRaffle(0, 1)
+      raffleStore.connect(rafflePlayer).enterRaffle(0, 1, {
+        value: ticketPrice.toString()
+      })
     ).to.be.satisfy
     
     await expect(
       raffleStore.connect(rafflePlayer).enterRaffle(0, 1)
-    ).to.be.revertedWith('Not enough tickets available')
+    ).to.be.revertedWith('Raffle no longer active')
   })
 
-  // TODO: test entering a closed / pending_completion raffle
+  it("Should check the eth deposit matches the ticket price", async () => {
+    await testNft.connect(raffleOwner).approve(raffleStore.address, nftId)
+    await raffleStore.connect(raffleOwner).createRaffle(testNft.address, nftId, totalRaffleTickets, totalRafflePrice);
+    
+    await expect(
+      raffleStore.connect(rafflePlayer).enterRaffle(0, 1)
+    ).to.be.revertedWith("Ticket price not paid")
 
-  // TODO: test ticket purchase transfers eth
+    await expect(
+      raffleStore.connect(rafflePlayer).enterRaffle(0, 1, {
+        value: ticketPrice.div(2)
+      })
+    ).to.be.revertedWith("Ticket price not paid")
+    
+    await expect(
+      raffleStore.connect(rafflePlayer).enterRaffle(0, 1, {
+        value: ticketPrice
+      })
+    ).to.emit(raffleStore, "TicketsPurchased").withArgs(0, rafflePlayer.address, 1);
 
+    await expect(
+      raffleStore.connect(rafflePlayer).enterRaffle(0, 2, {
+        value: ticketPrice.mul(2)
+      })
+    ).to.emit(raffleStore, "TicketsPurchased").withArgs(0, rafflePlayer.address, 2);
+
+    await expect(
+      raffleStore.connect(rafflePlayer).enterRaffle(0, 2, {
+        value: ticketPrice
+      })
+    ).to.be.revertedWith("Ticket price not paid")
+  })
+
+  it("Should choose a winner when the last ticket is purchased", async () => {
+    await testNft.connect(raffleOwner).approve(raffleStore.address, nftId)
+    await raffleStore.connect(raffleOwner).createRaffle(testNft.address, nftId, totalRaffleTickets, totalRafflePrice);
+    
+    // TODO: check link was sent to vrfCoordinator
+    // const LinkToken = await deployments.get('LinkToken')
+    // linkToken = await ethers.getContractAt('LinkToken', LinkToken.address)
+
+    // await expect(
+    //   raffleStore.connect(rafflePlayer).enterRaffle(0, totalRaffleTickets, {
+    //     value: ticketPrice.mul(totalRaffleTickets)
+    //   })
+    // ).to.changeTokenBalances(
+    //   linkToken,
+    //   [raffleStore, vrfCoordinatorMock],
+    //   [-200, 200]
+    // );
+
+    await raffleStore.connect(rafflePlayer).enterRaffle(0, totalRaffleTickets, {
+      value: ticketPrice.mul(totalRaffleTickets)
+    })
+
+    await expect(
+      (await raffleStore.raffles(0)).status
+    ).to.equal(1)
+
+  })
+
+  it("Should not sell tickets while the raffle is pending completion", async () => {
```

Comment:
Does this actually test purchasing a ticket while it's pending completion? Is it possible that the raffle would complete before the assertion, so we're actually testing purchasing a ticket for a completed raffle?